lib/order.rb

```diff
@@ -0,0 +1,27 @@
+require './lib/menu'
+
+class Order
+
+  attr_reader :basket, :menu, :total
+
+  def initialize
+    @basket = []
+    @menu = Menu.new
+    @total = 0
+  end
+
+  def add_dish(dish_name, quantity)
+    dish_to_add_to_basket = menu.dishes.select { |dish| dish[:name] == dish_name }.reduce(&:merge)
```

Comment:
Could 'dish_to_add_to_basket' have a more concise/easily re-useable name?

Nice one-liner syntax though, the code is easy to read.